diff --git a/main.js.old b/main.js
index dd13801..c244412 100644
--- a/main.js.old
+++ b/main.js
@@ -520,7 +520,13 @@ function main () {
                        return container.keys () .indexOf (element) > -1;
                }
                else {
-                       return container.indexOf (element) > -1;
+                       if (container.indexOf !== undefined) {
+                               return container.indexOf(element) > -1;
+                       } else if (container.has !== undefined) {
+                               return container.has(element);
+                       } else {
+                               return (element in container);
+                       }
                }
        }
        __all__.__in__ = __in__;
